src/protocol/spi/nrf24l01.c

```diff
@@ -434,6 +434,8 @@ u8 XN297_WritePayload(u8* msg, int len)
         packet[last++] = crc >> 8;
         packet[last++] = crc & 0xff;
     }
+    if (++last > 32)
```

Comment:
Weird, I've to add an extra byte to the emulated packet or the last byte of the packet (CRC low nibble) is replaced by CRC high nibble (analyzed the packets with SDR, eg expected CRC is 06EB but 0606 is transmitted). Seems it happens only with some specific packet lengths (10), there's something fishy here.

Reply:
I think I found the issue, seems like there is a race condition in NRF24L01_WritePayload, CS_HI() can be called before PROTOSPI_xfer has finished its work. Adding an usleep(5) just before CS_HI() fixes the problem. That's not elegant tho, I suppose we can check a flag to make sure the SPI transfer has completed instead ?